### PR TITLE
fixes xml_response fetch enumeration

### PR DESCRIPTION
**Pull request by martinheld**

### CODE REVIEW

File: lib/soapy_cake/admin_batched.rb [hunk 1]
```diff
@@ -35,7 +35,8 @@ def to_enum
             result = next_batch
             # raises StopIteration when less than `limit` elements are present
             # which is then rescued by `loop`
-            limit.times { y << result.next }
+            opts[:xml_response] ? y << result.next : limit.times { y << result.next }
```
Comment (dziemba) [inline lib/soapy_cake/admin_batched.rb]: Please don't use the ternary operator for statements that have side effects, it's hard to understand.

File: lib/soapy_cake/admin_batched.rb [hunk 2]
```diff
@@ -53,7 +54,7 @@ def next_batch
     def method_missing(name, method_opts = {}, limit = nil)
       fail Error, "Invalid method #{name}" unless ALLOWED_METHODS.include?(name)
 
-      BatchedRequest.new(admin, name, method_opts, limit).to_enum
+      BatchedRequest.new(admin, name, method_opts.merge(opts.slice(:xml_response)), limit).to_enum
```
Comment (dziemba) [inline lib/soapy_cake/admin_batched.rb]: Don't you rather want a new parameter for that? Does it really make sense to add this flag to the cake API opts?

**Comment by martinheld** [comment]
implemented

**Comment by dziemba** [comment]
:mushroom:

File: lib/soapy_cake/admin_batched.rb [hunk 3]
```diff
@@ -43,7 +40,18 @@ def to_enum
 
       private
 
-      def next_batch
+      def fetch_elements(enumerator)
```
Comment (dziemba) [inline lib/soapy_cake/admin_batched.rb]: `fetch_elements`? That doesn't sound any different than `fetch_batch`... Wouldn't `yield_elements` be better?

**Pull request merged**